samples/auth-user-sample/test/auth.test.js

```diff
@@ -0,0 +1,99 @@
+// /**
+//  * Copyright 2019 Google LLC
+//  *
+//  * Licensed under the Apache License, Version 2.0 (the "License");
+//  * you may not use this file except in compliance with the License.
+//  * You may obtain a copy of the License at
+//  *
+//  *   https://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  * Unless required by applicable law or agreed to in writing, software
+//  * distributed under the License is distributed on an "AS IS" BASIS,
+//  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  * See the License for the specific language governing permissions and
+//  * limitations under the License.
+//  */
+
+// 'use strict';
+
+const mocha = require('mocha');
+const describe = mocha.describe;
+const before = mocha.before;
+const after = mocha.after;
+const it = mocha.it;
+const sinon = require('sinon');
+require('chai').use(require('sinon-chai'));
+const {BigQuery} = require('@google-cloud/bigquery');
+const uuid = require('uuid');
+const proxyquire = require('proxyquire');
+
+const generateUuid = () => `gcloud-tests-${uuid.v4()}`.replace(/-/gi, '_');
+const datasetId = generateUuid();
+const tableId = generateUuid();
+let projectId;
+
+const bigquery = new BigQuery();
+
+describe('authUserFlow()', function() {
+  let readlineMock, generateUrlStub, questionStub, tokenStub, authUserFlow;
+
+  before(async () => {
+    await bigquery.createDataset(datasetId);
+    const [tableData] = await bigquery.dataset(datasetId).createTable(tableId);
+    projectId = tableData.metadata.tableReference.projectId;
+
+    questionStub = sinon.stub();
+
+    readlineMock = {
+      question: questionStub,
+    };
+
+    generateUrlStub = sinon.stub().returns('https://example.com');
+    tokenStub = sinon.stub().returns({tokens: 'tokens'});
+
+    authUserFlow = proxyquire('../authUserFlow.js', {
+      'google-auth-library': {
+        OAuth2Client: sinon.stub().callsFake(() => {
+          return {
+            generateAuthUrl: generateUrlStub,
+            getToken: tokenStub,
+          };
+        }),
+      },
+    });
+  });
+  after(async () => {
+    await bigquery
+      .dataset(datasetId)
+      .delete({force: true})
+      .catch(console.warn);
+  });
+
+  it('should exchange code for tokens', async function() {
+    const output = await authUserFlow.main.exchangeCode('abc123');
+    sinon.assert.match(output, 'tokens');
```

Comment:
I'm a little confused about the usage of `sinon.assert` here.  How is this different from a regular `assert`?

Reply:
The docs for testing whether stubs were called use sinon.assert, so I went with that. I haven't seen how to do the same with regular assert.

Reply:
I think that this is equivalent to:

```js
assert.strictEqual(output, 'tokens');
```

Took me for a slight loop too, just because I was trying to figure out whether sinon was doing something fancy, e.g., spying on `output` or something. I'd probably opt for `assert` like @JustinBeckwith suggests, because it communicates the fact that this is just a simple assertion.

Reply:
Yeah, this is correct, I thought this was asking about using sinon.assert when checking to see if a stub was called. I misread. Should I be using assert for these simple comparisons and sinon.assert for checking stubs? Or is there another way to use assert to just verify that stubbed out portions of the authorization flow are being called?

Reply:
As a general rule, I'd use Sinon's methods for checking if methods are called, which it's great at, e.g.,

`sinon.assert.called(authenticateStub);` which seems like great use of sinon's assertions. For simple comparisons, I'd probably opt for `assert()`, `assert.strictEqual()`, etc.

Reply:
Gotcha. Thanks.